packages/core/src/repositories/evaluationsRepository.ts

```diff
@@ -1,29 +1,41 @@
 import { omit } from 'lodash-es'
 
-import { and, eq, getTableColumns, inArray, isNull, sql } from 'drizzle-orm'
+import { and, eq, getTableColumns, inArray, isNull } from 'drizzle-orm'
 
 import { EvaluationDto } from '../browser'
-import { EvaluationMetadataType } from '../constants'
+import { EvaluationMetadataType, EvaluationResultableType } from '../constants'
 import { NotFoundError, PromisedResult, Result } from '../lib'
 import {
   connectedEvaluations,
+  evaluationConfigurationBoolean,
+  evaluationConfigurationNumerical,
+  evaluationConfigurationText,
   evaluationMetadataLlmAsJudgeAdvanced,
+  evaluationMetadataLlmAsJudgeSimple,
   evaluations,
 } from '../schema'
+import { getSharedTableColumns } from '../schema/schemaHelpers'
 import RepositoryLegacy from './repository'
 
 const tt = {
   ...getTableColumns(evaluations),
-  metadata: {
-    id: sql<number>`llm_as_judge_evaluation_metadatas.id`.as(
-      'metadata_metadata_id',
-    ),
-    ...omit(getTableColumns(evaluationMetadataLlmAsJudgeAdvanced), [
-      'id',
-      'createdAt',
-      'updatedAt',
-    ]),
-  },
+  metadata: omit(
+    getSharedTableColumns(evaluations.metadataType, {
+      [EvaluationMetadataType.LlmAsJudgeAdvanced]:
+        evaluationMetadataLlmAsJudgeAdvanced,
+      [EvaluationMetadataType.LlmAsJudgeSimple]:
+        evaluationMetadataLlmAsJudgeSimple,
+    }),
+    ['id', 'createdAt', 'updatedAt'],
+  ),
+  resultConfiguration: omit(
+    getSharedTableColumns(evaluations.resultType, {
+      [EvaluationResultableType.Boolean]: evaluationConfigurationBoolean,
+      [EvaluationResultableType.Number]: evaluationConfigurationNumerical,
+      [EvaluationResultableType.Text]: evaluationConfigurationText,
+    }),
+    ['id', 'createdAt', 'updatedAt'],
```

Comment:
Here, the `metadata` attribute contains either the attributes from `evaluationMetadataLlmAsJudgeAdvanced` or `evaluationMetadataLlmAsJudgeSimple`, depending on the value of `evaluations.metadataType`. The `getSharedTableColumns` returns the correct SQL query to return the attributes from either one or the other depending on the actual value!

The same happens with `resultConfiguration`, which can be the attributes from `evaluationConfigurationBoolean`, `evaluationConfigurationNumerical` or `evaluationConfigurationText`, depending on the value of `evaluations.resultType`.